waku.md

```diff
@@ -81,6 +81,10 @@ pow-requirement  = pow
 ; bloom filter is "a byte array"
 bloom-filter     = bytes
 
+; List of topics interested in
+; Bounded to maximum of 100 topics
```

Comment:
whats a good number here? @adambabik @kdeme

Reply:
Yeah, I don't know. Some value > than the maximum of chats the app allows?

Reply:
I'm not sure that's limited, and if it is, is there any risk it is higher than what an envelope can fit? :P

@cammellos any thoughts?

Reply:
depends for what is used exactly, if it's just for historical request, we can get away with requesting multiple times.
If it's for just receiving messages, this is likely to be too little, currently we use:

1 topic per public chat
2/3 topics per contact/member of a group chat/private chat (1 contact code topic, 1 negotiated topic, 1 partitioned topic)
2 discovery topics (partitioned, personal)

Of this we could do some topic hygiene:
1) Contact code topic can be probably fetched "on-demand", so you only look for a contact code at intervals or when you want to contact the user
2) Negotiated topic is necessary if we want to keep them
3) Partitioned topic is only used because we need to listen to any topic that we publish to, I assume we are going to lift this requirement in waku? if so we don't need this.

100 definitely seems too little, I have last time I checked my desktop app would be using ~250 topics, although my usage is a bit uncommon as I test stuff, but I would go for a fairly high number to start with (say 10k, doesn't matter), as currently is unbound.

What is exactly the concern here? Performance/security or there are some technical constraints?

Reply:
If there are some tech constraints there might be an argument for still using a bloom filter, although of much larger size (say 1KB, which would be around 256 topics in clear), making clashes unlikely for most cases (say n topics < 500), while keeping straight topics for historical requests (most bandwidth usage on mobile comes from mailserver interactions likely, if we assume they are offline more than 50% of the day and traffic is equally distributed during the day, which is not strictly true, but for simplicity we can assume so)

Reply:
>  If it's for just receiving messages, this is likely to be too little, currently we use:

It is meant for receiving messages, yes.

> What is exactly the concern here? Performance/security or there are some technical constraints?

One, I don't want us to have unbound things in general. Second, specifically, I suppose we want to stay within a reasonable payload size. Which would point to something like 4 bytes*256=~1KB as you say.

Issue with bloom filter is still false positives. Which would need to get arbitrarily small as network grows. I suppose this could be dynamically adjusted, but seems complex to me.

1) If we set it to 256/512/1000, would that be enough for non-perverse Core app needs? Do people see an issue with 1/2/4KB sized payloads?

2) If we split it somehow, what would be a reasonable strategy for this? E.g. [1/2, topics], [2/2 topics cont]

3) As a fallback when N>256 we can fallback to bloom filter perhaps, with the concession that this might impact bandwidth usage.

Reply:
Re payload size, in RLPx v5 https://eips.ethereum.org/EIPS/eip-706 was introduced. In it they allude to up to 16MB message size (presumably for `eth` etc protocols). This is different from envelopes. So I suppose setting max to 10k would be OK.

Any objections? @adambabik @kdeme

Reply:
Yeah that is the frame-size that is referred to: https://github.com/ethereum/devp2p/blob/master/rlpx.md#message-encoding

In nim-eth a check is done if it is not bigger than 10MB actually, I though the same check was done in geth, but perhaps that was on the "unpacked" protocol message, I'd have to check.
In Whisper we currently limit it to 1MB, but this is nowhere specified so we could increase.

Reply:
10k would be ~40kb I believe (without compression), so we should be fine there for now I think